src/modules/blob.js

```diff
@@ -0,0 +1,86 @@
+import { encodeAddress, randomAsHex } from '@polkadot/util-crypto';
+import { getSignatureFromKeyringPair, getStateChange } from '../utils/misc';
+import NoBlobError from '../utils/errors/no-blob-error';
+
+export const DockBlobMethod = 'dock';
+export const DockBlobQualifier = `blob:${DockBlobMethod}:`;
+export const DockBlobIdByteSize = 32;
+
+/**
+ * Create and return a fully qualified Dock Blob, i.e. "did:dock:<SS58 string>"
+ * @returns {string} - The Blob
+ */
+export function createNewDockBlobId() {
+  const hexId = randomAsHex(DockBlobIdByteSize);
+  return blobHexIdToQualified(hexId);
+}
+
+/**
+ * Return a fully qualified Dock Blob, i.e. "did:dock:<SS58 string>"
+ * @returns {string} - The Blob
+ */
+export function blobHexIdToQualified(hexId) {
+  const ss58Id = encodeAddress(hexId);
+  return `${DockBlobQualifier}${ss58Id}`;
+}
+
+/** Class to create and update Blobs on chain. */
+class BlobModule {
+  /**
+   * Creates a new instance of BlobModule and sets the api
+   * @constructor
+   * @param {object} api - PolkadotJS API Reference
+   */
+  constructor(api) {
+    this.api = api;
+    this.module = api.tx.blobStore;
+  }
+
+  /**
+   * Register a new Blob on the Dock Chain
+   * @param {object} blob - struct to store on chain
+   * @param {object} keyPair - Key pair to sign with
+   * @return {object} The extrinsic to sign and send.
+   */
+  new(blob, keyPair) {
```

Comment:
Same comment as i had made to revocation methods, it should be possible to pass in signature from outside (hardware wallet, HSM, etc)

Reply:
Kept the keypair option and allowed signature to be sent too.